app/dao/users_dao.py

```diff
@@ -28,15 +28,16 @@ def save_user_attribute(usr, update_dict={}):
     db.session.commit()
 
 
-def save_model_user(usr, update_dict={}, pwd=None):
-    if pwd:
-        usr.password = pwd
-        usr.password_changed_at = datetime.utcnow()
+def save_model_user(user, update_dict={}, password=None):
+    if password:
+        user.password = password
+        user.password_changed_at = datetime.utcnow()
+        user.email_access_last_validated_at = datetime.utcnow()
```

Comment:
I'm not sure if we should have this line of code here for the following reason.

This function is `save_model_user` which may be used in varying places around our codebase without it being aware that if you change the password then you also set the `email_access_last_validated_at` unless you look at the code. This means there could be a risk of us bumping the last validated at date without being aware we are doing it and may not want to do so.

At the moment it isn't causing an issue with our existing codebase, the functionality seems to work as expected, however I feel it adds some opaque behaviour to the `save_model_user` function. 

I think I like how you've done it for `update_user_password(user, password, from_email=False)` with the `from_email` param, although I'm not sure I'd suggest doing it that way for this function itself. 

I'm wondering if we should make it super explicit what we are doing by adding a function to be called in the view code like `validate_email_access` which then each view would be responsible for deciding whether email access has been used.

I'm not 100% on this approach so would be keen to hear if you have any thoughts on potential approaches (including this one) and pros and cons.